src/Games/NexusMods.Games.AdvancedInstaller.UI/Content/Right/SelectLocation/SelectLocationViewModel.cs

```diff
@@ -0,0 +1,199 @@
+
+        // Add all the top level game locations to suggested entries.
+        foreach (var (locationId, fullPath) in register.GetTopLevelLocations())
+        {
+            suggestedEntries.Add(new SuggestedEntryViewModel(
+                Guid.NewGuid(),
+                fullPath,
+                locationId,
+                new GamePath(locationId, RelativePath.Empty)));
+
+            // Add nested locations to suggested entries.
+            foreach (var nestedLocation in register.GetNestedLocations(locationId))
+            {
+                var nestedFullPath = register.GetResolvedPath(nestedLocation);
+                var relativePath = nestedFullPath.RelativeTo(fullPath);
+                suggestedEntries.Add(new SuggestedEntryViewModel(
+                    Guid.NewGuid(),
+                    nestedFullPath,
+                    nestedLocation,
+                    new GamePath(locationId, relativePath)));
+            }
+        }
+
+        return suggestedEntries;
+    }
+
+    /// <summary>
+    /// Creates the tree entries from the LocationIds and potentially the Loadout folder structure.
+    /// </summary>
+    /// <param name="register">The game locations register</param>
+    /// <param name="loadout">The loadout, can be null.</param>
+    /// <returns>The list of created tree entries that need to be added to the cache.</returns>
+    private static List<ISelectableTreeEntryViewModel> CreateTreeEntries(GameLocationsRegister register,
+        Loadout? loadout)
+    {
+        // Initial population of the tree based on LocationIds
+        List<ISelectableTreeEntryViewModel> treeEntries = new();
+
+        foreach (var (locationId, fullPath) in register.GetTopLevelLocations())
+        {
+            var treeEntry = new SelectableTreeEntryViewModel(
+                new GamePath(locationId, RelativePath.Empty),
+                SelectableDirectoryNodeStatus.Regular);
+
+            var createFolderEntry = new SelectableTreeEntryViewModel(
+                new GamePath(locationId, ISelectableTreeEntryViewModel.CreateFolderEntryName),
+                SelectableDirectoryNodeStatus.Create);
+
+            treeEntries.Add(treeEntry);
+            treeEntries.Add(createFolderEntry);
+
+            // Add nested
+            foreach (var nestedLocation in register.GetNestedLocations(locationId))
+            {
+                var nestedFullPath = register.GetResolvedPath(nestedLocation);
+                var relativePath = nestedFullPath.RelativeTo(fullPath);
+                var relativeGamePath = new GamePath(locationId, relativePath);
+                // Add all nodes from root to nested location.
+                treeEntries.AddRange(CreateMissingEntriesForGamePath(treeEntries.ToArray(), relativeGamePath, false));
+            }
+        }
+
+        if (loadout != null)
+        {
+            // TODO: Potentially add entries to the tree to represent all the folders found in the loadout.
+        }
```

Comment:
Is this still relevant?

Reply:
Yes, as potential future improvement. I would make it wait for all the event based loadout changes.